src/uptane/initialize.cc

```diff
@@ -128,28 +128,25 @@ bool Repository::initEcuKeys(const UptaneConfig& uptane_config) {
     primary_public = p11.getUriPrefix() + uptane_config.public_key_path;
     primary_private = p11.getUriPrefix() + uptane_config.private_key_path;
     std::string public_key_content;
-    if (!p11.readPublicKey(uptane_config.public_key_path, &public_key_content)) {
-      LOGGER_LOG(LVL_error, "Public key with id " << uptane_config.private_key_path << " was not found");
-      return false;
+    if (p11.readPublicKey(uptane_config.public_key_path, &public_key_content)) {
+      primary_public_id = Crypto::getKeyId(public_key_content);
+      setEcuKeysMembers(primary_public, primary_private, primary_public_id, kPkcs11);
+      return true;
     }
-    primary_public_id = Crypto::getKeyId(public_key_content);
-
-    setEcuKeysMembers(primary_public, primary_private, primary_public_id, kPkcs11);
-    return true;
+    if (!p11.generateRSAKeyPair(uptane_config.private_key_path)) return false;
```

Comment:
OK, I can see where the problem is. After we generate keys with generateRSAKeyPair we don't initialize members. Correct logic would be:
1. Try to readPublicKey.
2. On success skip, otherwise generateRSAKeyPair.
3. readPublicKey, on success set primary_public_id and setEcuKeyMembers, otherwise fail.

Can't make a patch right now unfortunately, let's do that tomorrow.